files/zh-cn/web/javascript/reference/global_objects/set/@@iterator/index.md

```diff
@@ -5,60 +5,73 @@ slug: Web/JavaScript/Reference/Global_Objects/Set/@@iterator
 
 {{JSRef}}
 
-**`@@iterator`** 属性的初始值和 {{jsxref("Set.prototype.values()", "values")}} 属性的初始值是同一个函数。
+{{jsxref("Set")}} 实例的 **`[@@iterator]()`** 方法实现了[可迭代协议](/zh-CN/docs/Web/JavaScript/Reference/Iteration_protocols)以允许 `Set` 对象被大多数期望可迭代对象的语法所消费，例如[展开语法](/zh-CN/docs/Web/JavaScript/Reference/Operators/Spread_syntax)和 {{jsxref("Statements/for...of", "for...of")}} 循环。它返回一个[集合迭代器对象](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Iterator)，它会以插入顺序生成集合的值。
+
+该属性的初始值与 {{jsxref("Set.prototype.values()")}} 属性的初始值是同一个函数对象。
 
 {{EmbedInteractiveExample("pages/js/set-prototype-@@iterator.html")}}
 
-## Syntax
+## 语法
 
-```plain
-mySet[Symbol.iterator]
+```js-nolint
+set[Symbol.iterator]()
 ```
 
-### Return value
+### 参数
+
+无。
+
+### 返回值
 
-返回 `Set` **iterator** 函数，默认值是 {{jsxref("Set.prototype.values()", "values()")}} 函数。
+与 {{jsxref("Set.prototype.values()")}} 返回值相同：一个新的[可迭代迭代器对象](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Iterator)，它会以插入顺序生成集合的值。
 
-## Examples
+## 示例
 
-### Using `[@@iterator]()`
+### 使用 for...of 循环进行迭代
+
+请注意，通常你不需要直接调用此方法。`@@iterator` 方法的存在使得 `Set` 对象[可迭代](/zh-CN/docs/Web/JavaScript/Reference/Iteration_protocols#可迭代协议)，而像 `for...of` 循环这样的迭代语法会自动调用此方法以获取用于循环的迭代器。
 
 ```js
 const mySet = new Set();
 mySet.add("0");
 mySet.add(1);
 mySet.add({});
 
-const setIter = mySet[Symbol.iterator]();
-
-console.log(setIter.next().value); // "0"
-console.log(setIter.next().value); // 1
-console.log(setIter.next().value); // Object
+for (const v of mySet) {
+  console.log(v);
+}
 ```
 
-### Using `[@@iterator]()` with `for..of`
+### 手动控制迭代器
+
+你仍然可以手动调用返回的迭代器对象的 `next()` 方法来获得最大程度的控制权。
 
 ```js
 const mySet = new Set();
 mySet.add("0");
 mySet.add(1);
 mySet.add({});
 
-for (const v of mySet) {
-  console.log(v);
-}
+const setIter = mySet[Symbol.iterator]();
+
+console.log(setIter.next().value); // "0"
+console.log(setIter.next().value); // 1
+console.log(setIter.next().value); // Object
```

Comment:
Could you update the content in en-US. I've seem a document that has used "{}" rather than "object"

```suggestion
console.log(setIter.next().value); // {}
```

Reply:
Okay.

Reply:
See https://github.com/mdn/content/pull/29586.